core/common/src/main/java/alluxio/worker/block/io/BlockReader.java

```diff
@@ -46,4 +48,18 @@
    * @return channel
    */
   ReadableByteChannel getChannel();
+
+  /**
+   * Transfers data from this reader to the buffer.
+   *
+   * @param buf the byte buffer
+   * @return the number of bytes transferred, -1 if the end of the block is reached
+   * @throws IOException if any I/O errors occur
+   */
+  int transferTo(ByteBuf buf) throws IOException;
```

Comment:
I found it becomes confusing in our code to have ByteBuf and ByteBuffer mixed (one in read func, the other in transferTo func). Very easy to make mistakes as these two buffers have similar but different semantics. Is this only to ease the netty callers? can we always use ByteBuffer and convert ByteBuffer to ByteBuf outside the block reader/writer?